### PR TITLE
Build Python wheels with tag manylinux2010

### PR DESCRIPTION
**Pull request by lidizheng**
Let's see if the tests are happy.

### CODE REVIEW

File: tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Install Python build requirements
+RUN /opt/python/cp27-cp27m/bin/pip install cython
+RUN /opt/python/cp27-cp27mu/bin/pip install cython
+RUN /opt/python/cp34-cp34m/bin/pip install cython
+RUN /opt/python/cp35-cp35m/bin/pip install cython
+RUN /opt/python/cp36-cp36m/bin/pip install cython
+RUN /opt/python/cp37-cp37m/bin/pip install cython
+
+####################################################
+# Install auditwheel with fix for namespace packages
```
Comment (gnossen) [inline tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile]: Is this necessary?

**Comment by jtattermusch** [comment]
Would this by any chance help with the problem described here: https://github.com/grpc/grpc/pull/20100#issuecomment-532593758  (it's preventing us from upgrading third_party/boringssl)?

**Comment by lidizheng** [comment]
@jtattermusch As commented in #20100, this will be helpful eventually. But for now, we won't be able to remove `manylinux1`.

**Comment by lidizheng** [comment]
@jtattermusch PTALA. All failures are not-related to this PR.

File: tools/run_tests/artifacts/build_artifact_python.sh [hunk 2]
```diff
@@ -79,12 +79,12 @@ ${SETARCH_CMD} "${PYTHON}" tools/distrib/python/grpcio_tools/setup.py bdist_whee
 if [ "$GRPC_BUILD_MANYLINUX_WHEEL" != "" ]
 then
   for wheel in dist/*.whl; do
-    "${AUDITWHEEL}" show "$wheel" | tee /dev/stderr | grep \"manylinux1
+    "${AUDITWHEEL}" show "$wheel" | tee /dev/stderr |  grep -E -w 'manylinux(1|2010)_(x86_64|i686)'
```
Comment (gnossen) [inline tools/run_tests/artifacts/build_artifact_python.sh]: Nice change!

File: tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile (hunk 1 above)
Comment (lidizheng) [reply tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile]: For infrastructure code, I think pin the version of auditwheel is better than always use the latest version.

Comment (gnossen) [reply tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile]: Agree. The question is whether the install from a git repository is still necessary or not.

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/20306 https://github.com/grpc/grpc/issues/20307 https://github.com/grpc/grpc/issues/20308 https://github.com/grpc/grpc/issues/20309 https://github.com/grpc/grpc/issues/20310

File: tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile (hunk 1 above)
Comment (lidizheng) [reply tools/dockerfile/grpc_artifact_python_manylinux2010_x64/Dockerfile]: I will investigate and see if I can remove all of the hack, or part of the hack 👌

**Pull request merged**